Implementing a Write Operation With an Different Output (#1687)
api-platform/docs: core/dto.md

    }
}
```

## Implementing a Write Operation With an Output Different From the Resource

> The logic stay almost the same than for a [State Provider](./state-providers.md).

Reviewer: ```suggestion
The logic stays almost the same than for a [State Provider](./state-providers.md).
```
I don't think this sentence is necessary though.

Author: I deleted this sentence.

For returning another representation of your data in a [State Processor](./state-processors.md), you should specify your processor class in the `processor` attribute and same for your `output`.

```php
<?php

namespace App\Entity;

use ApiPlatform\Metadata\Post;
use App\Dto\AnotherRepresentation;
use App\State\BookRepresentationProcessor;

#[Post(output: AnotherRepresentation::class, processor: BookRepresentationProcessor::class)]
class Book {}
```

> Here the `$data` attribute represent an instance of your ressource. 

Reviewer: ```suggestion

Here the `$data` attribute represents an instance of your resource.

```

```php
<?php

namespace App\State;

use App\Dto\AnotherRepresentation;
use App\Model\Book;
use ApiPlatform\Metadata\Operation;

> use ApiPlatform\State\ProcessorInterface;

Reviewer: Please add them before.

Author: In the `Provider` example, there are imported before.

<img width="1022" alt="Capture d’écran 2023-01-15 à 12 57 40" src="https://user-images.githubusercontent.com/62953579/212539200-6d5b3364-d6a2-4c43-ad83-682867ccfbb8.png">

It's a mistake or not? We should stay coherent 😬

Reviewer: It's a mistake.

final class BookRepresentationProcessor implements ProcessorInterface
{
     /**
     * @param Book $data
     */
    public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
    {
        return new AnotherRepresentation(
            $data->getId(),
            $data->getTitle(),

>             // etc

Reviewer: ```suggestion
            // etc.
```

For returning another representation of your data in a [State Processor](./state-processors.md), you should specify your processor class in the `processor` attribute and same for your `output`.

[codeSelector]

```php
<?php

namespace App\Entity;

use ApiPlatform\Metadata\Post;
use App\Dto\AnotherRepresentation;
use App\State\BookRepresentationProcessor;

#[Post(output: AnotherRepresentation::class, processor: BookRepresentationProcessor::class)]
class Book {}
```
```yaml
# api/config/api_platform/resources.yaml
App\Entity\Book:
    operations:
        ApiPlatform\Metadata\Post:
            output: App\Dto\AnotherRepresentation
            processor: App\State\BookRepresentationProcessor
```
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<!-- api/config/api_platform/resources.xml -->

<resources xmlns="https://api-platform.com/schema/metadata/resources-3.0"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0
        https://api-platform.com/schema/metadata/resources-3.0.xsd">
    <resource class="App\Entity\Book">
        <operations>

>             <operation class="ApiPlatform\Metadata\Post" processor="App\State\BookRepresentationProcessor"

Reviewer: Maybe add `processor` below?

Author: done
